### PR TITLE
Fix errors in UI automated test caused by upgrade to 2024.1

### LINKED ISSUE #740: Upgrade to IntelliJ IDEA 2024.1: Investigate and resolve issues with UI test automation.
See comments here: https://github.com/OpenLiberty/liberty-tools-intellij/issues/722#issuecomment-2093670568.

The `runIde` crashes: https://github.com/OpenLiberty/liberty-tools-intellij/issues/723 may be contributing to failures. Other updates may be required to the tests due to changes in the UI in IntelliJ 2024.1.

### PR DESCRIPTION
**Pull request by turkeylurkey**
I included changes to build.gradle for 2024.1. I assume we can rationalize these version numbers during the merge process.

Fixes #740

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -137,7 +137,7 @@ runIde {
 
 intellij {
     // For a full list of IntelliJ IDEA releases please see https://www.jetbrains.com/intellij-repository/releases
-    version = '2023.3.4'
+    version = '2024.1'
```
Comment (mrglavas) [inline build.gradle:140]: We should target 2024.1.1 which seems to resolve the `runIde` crash that was occurring on Windows when opening a new terminal.

Comment (turkeylurkey) [reply build.gradle]: OK

File: build.gradle [hunk 2]
```diff
@@ -81,7 +81,7 @@ dependencies {
     }
     //required by lsp4j as the version from IJ is incompatible
 //    implementation 'com.google.code.gson:gson:2.8.9'
-//    implementation 'com.vladsch.flexmark:flexmark:0.50.50'
+    implementation 'com.vladsch.flexmark:flexmark-all:0.64.8'
```
[the remark this answers is not in the document]
Comment (turkeylurkey) [reply build.gradle]: Correct, removed.

**Review by mrglavas — APPROVED** [review]
Looks good. Thanks.

**Pull request merged**